enterprise/app/services/enterprise/action_service.rb

```diff
@@ -1,10 +1,15 @@
 module Enterprise::ActionService
-  def add_sla(sla_policy)
+  def add_sla(sla_policy_id)
+    sla_policy = @account.sla_policies.find(sla_policy_id.first)
+    Rails.logger.info "SLA:: Adding SLA #{sla_policy.id} to conversation: #{@conversation.id}"
     @conversation.update!(sla_policy_id: sla_policy.id)
     create_applied_sla(sla_policy)
   end
 
   def create_applied_sla(sla_policy)
+    return if AppliedSla.exists?(conversation_id: @conversation.id, account_id: @conversation.account_id, sla_policy_id: sla_policy.id)
```

Comment:
we won't need this check if conversation.sla_policy? is present in the above check right ?

Reply:
True. Will make the change.